src/modrover/rover.py

```diff
@@ -113,6 +169,102 @@ def explore(self, dataset: pd.DataFrame, strategy: Union[str, RoverStrategy]):
     def _fit_layer(self, dataset: pd.DataFrame, learners: list[Learner]) -> None:
         """Fit a layer of models. Store results on the learners dict."""
         for learner in learners:
-            print(learner)
```

Comment:
prints aren't good practice in production code, but it might be worth adding a logging library to let the user know about progress

Reply:
Yes, it is probably good to chat about the workflow of `rover` from a user perspective, we can also borrow what is happening in scikit-learn, if they have any loggers in their model.

Reply:
Is there a reason that we need to have `_fit_layer` function compare to directly write it out in the `_explore` function?

Reply:
I have that extra method just because in the future we might want to implement a more sophisticated parallelism model than a single for loop. If we want to implement concurrency via multiprocessing or joblib think it'd slot nicely in here